Tests/OpenCombineTests/PublisherTests/FutureTests.swift

```diff
@@ -0,0 +1,117 @@
+//
+//  FutureTests.swift
+//  
+//
+//  Created by Max Desiatov on 24/11/2019.
+//
+
+import XCTest
+
+#if OPENCOMBINE_COMPATIBILITY_TEST
+import Combine
+#else
+import OpenCombine
+#endif
+
+@available(macOS 10.15, iOS 13.0, *)
+final class FutureTests: XCTestCase {
+    enum TestFailureCondition: Error {
+        case anErrorExample
+    }
+
+    /// example of an asynchronous function to be called from within a Future and its
+    /// completion closure
+    func asyncAPICall(
+        sabotage: Bool,
+        completion completionBlock: @escaping ((Bool, Error?) -> Void)
+    ) {
+        DispatchQueue.global(qos: .background).async {
+            let delay = Int.random(in: 1...3)
+            print(" * making async call (delay of \(delay) seconds)")
+            sleep(UInt32(delay))
+            if sabotage {
+                completionBlock(false, TestFailureCondition.anErrorExample)
+            }
+            completionBlock(true, nil)
+        }
+    }
```

Comment:
I think we should avoid asynchronous testing: it makes tests flaky and slow.

For testing `Future`, the following scenario should suffice:

1. Initialize the future, store the promise in a local variable.
1. After initialization, ensure that the future has not yet received a value and a completion.
1. Resolve the promise using the local variable it's been saved to.
1. Ensure that the future did receive a value and a completion.

This is what I love about Combine: it abstracted away all the asynchrony behind `Scheduler`s, so we can easily test our publishers without turning to `DispatchQueue`s.